### PR TITLE
Profiles operation are added

### PR DESCRIPTION
**Pull request by nishith-vihar**
Profile operations are added in the profile_ops library
and an example test case is also added and changes are made in
the mixin.py file to incorporate the profile_ops.

Fixes: #381
Signed-off-by: Nishith Vihar Sakinala <nsakinal@redhat.com>

### CODE REVIEW

File: common/ops/gluster_ops/profile_ops.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        Example:
+            profile_start(node, "testvol")
+        """
+        cmd = f"gluster volume profile {volname} start"
+
+        ret = self.execute_abstract_op_node(cmd, node)
+
+        return ret
+
+    def profile_info(self, volname: str, node: str,
```
Comment (srijan-sivakumar) [inline common/ops/gluster_ops/profile_ops.py:41]: nit: Let's keep node as the last parameter. This is sort of being followed in most of the ops in redant.

**Review by srijan-sivakumar — COMMENTED** [review]
Just a small change (nit). Rest LGTM.

File: common/ops/gluster_ops/profile_ops.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            profile_stop(node, "testvol")
+        """
+        cmd = f"gluster volume profile {volname} stop"
+
+        ret = self.execute_abstract_op_node(cmd, node)
+
+        return ret
+
+    def check_profile_options(self, options: str) -> bool:
+        """Helper function to valid if profile options.
```
Comment (aujjwal-redhat) [inline common/ops/gluster_ops/profile_ops.py:96]: validate the profile option

Comment (nishith-vihar) [reply common/ops/gluster_ops/profile_ops.py:96]: Its already being called from profile_info

File: common/ops/gluster_ops/profile_ops.py (hunk 1 above)
Comment (nishith-vihar) [reply common/ops/gluster_ops/profile_ops.py:41]: Ack

File: common/ops/gluster_ops/profile_ops.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    Profile Ops class provides APIs to perform profile
+    operations like profile_start, profile_stop, profile_info,
+    check_profile_options etc.
+    """
+
+    def profile_start(self, volname: str, node: str) -> dict:
+        """
+        Start profile on the specified volume.
+        Args:
+            volname (str): Volume on which profile has to started.
```
[the remark this answers is not in the document]
Comment (nishith-vihar) [reply common/ops/gluster_ops/profile_ops.py:21]: Ack

File: common/ops/gluster_ops/profile_ops.py (hunk 1 above)
Comment (nishith-vihar) [reply common/ops/gluster_ops/profile_ops.py:41]: Actually the optional parameters have to be at the end right. node is not optional over here

File: common/ops/gluster_ops/profile_ops.py (hunk 2 above)
Comment (aujjwal-redhat) [reply common/ops/gluster_ops/profile_ops.py:95]: Hey, I meant the docstring. XD

File: common/ops/gluster_ops/profile_ops.py (hunk 1 above)
Comment (srijan-sivakumar) [reply common/ops/gluster_ops/profile_ops.py]: cool

File: common/ops/gluster_ops/profile_ops.py (hunk 2 above)
Comment (srijan-sivakumar) [reply common/ops/gluster_ops/profile_ops.py:95]: @nishith-vihar

Comment (nishith-vihar) [reply common/ops/gluster_ops/profile_ops.py:95]: Yeah got it XD

**Comment by srijan-sivakumar** [comment]
cool. And the test results :D

**Comment by nishith-vihar** [comment]
Screenshot from 2021-05-19 18-44-52

**Review by srijan-sivakumar — APPROVED** [review]
LGTM

**Pull request merged**